gm4_combat_expanded/data/gm4_combat_expanded/functions/weapon/update_sharpness.mcfunction

```diff
@@ -0,0 +1,38 @@
+# update the weapon damage tooltip when sharpness level changes
+# @s = player wielding the weapon
+# at @s
+# run from weapon/process
+
+# store actual sharpness level in storage
+execute store result storage gm4_combat_expanded:temp tag.gm4_combat_expanded.sharpness short 1 run scoreboard players get $current_sharpness gm4_ce_data
+
+# check if text should be light green
+execute store result score $damage_buffed gm4_ce_data run data get storage gm4_combat_expanded:temp tag.gm4_combat_expanded.damage_buffed
+
+# get weapon damage and added damage from sharpness (0.5 * level + 0.5)
+execute store result score $damage gm4_ce_data run data get storage gm4_combat_expanded:temp tag.AttributeModifiers[{Name:"gm4_ce_base_attack_damage"}].Amount 10
+scoreboard players operation $current_sharpness gm4_ce_data *= #5 gm4_ce_data
+execute if score $current_sharpness gm4_ce_data matches 1.. run scoreboard players add $current_sharpness gm4_ce_data 5
+scoreboard players operation $damage gm4_ce_data += $current_sharpness gm4_ce_data
+
+# get digits seperate
+scoreboard players operation $damage_10 gm4_ce_data = $damage gm4_ce_data
+scoreboard players operation $damage_10 gm4_ce_data /= #10 gm4_ce_data
+scoreboard players operation $damage_01 gm4_ce_data = $damage gm4_ce_data
+scoreboard players operation $damage_01 gm4_ce_data %= #10 gm4_ce_data
+
+# get new lore lines
+# slots 0 is reserved for the weapon, slot 1 is reserved for the modifier
+loot replace block 29999998 1 7134 container.2 loot gm4_combat_expanded:technical/sharpness_lore
+
+# update lore with lib_lore
+data modify storage gm4_lore:temp Source set from storage gm4_combat_expanded:temp tag.display.Lore
+data modify storage gm4_lore:temp Target set value '{"translate":"%1$s%3427655$s","with":[{"italic":false,"color":"gray","text":"When in Main Hand:"},{"translate":"item.gm4.combat_expanded.lore.mainhand"}]}'
```

Comment:
Left a note in one of the loot tables; this is already a translated string in vanilla and thus the vanilla translation should be used. That means this Target needs to be updated as well